web/src/rendered_markdown.js

```diff
@@ -72,12 +72,26 @@ export function set_name_in_mention_element(element, name) {
     }
 }
 
+function is_client_safari() {
+    // Since GestureEvent is only supported on Safari, we can use it
+    // to detect if the browser is Safari including Safari on iOS.
+    // https://developer.mozilla.org/en-US/docs/Web/API/GestureEvent
+    return Boolean(window.GestureEvent);
+}
```

Comment:
We should put this in `util.ts` next to this function:

```
// check by the userAgent string if a user's client is likely mobile.                                   
export function is_mobile(): boolean {                                                                  
    const regex = "Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini";                      
    return new RegExp(regex, "i").test(window.navigator.userAgent);                                     
}                                                                                                       
```

Reply:
Done